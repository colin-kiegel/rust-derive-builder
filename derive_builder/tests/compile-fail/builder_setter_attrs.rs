#![deny(unused_must_use)]

#[macro_use]
extern crate derive_builder;

#[derive(Debug, PartialEq, Default, Builder, Clone)]
pub struct Lorem {
    ok: String,

    // This foolish repr() attribute generates an unused attribute warning
    #[builder_setter_attrs(must_use)]
    broken: usize,
}

fn main() {
    LoremBuilder::default().broken(42);
}
