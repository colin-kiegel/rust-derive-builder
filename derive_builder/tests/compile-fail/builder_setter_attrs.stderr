error: unused return value of `LoremBuilder::broken` that must be used
  --> tests/compile-fail/builder_setter_attrs.rs:18:5
   |
18 |     LoremBuilder::default().broken(42);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> tests/compile-fail/builder_setter_attrs.rs:1:9
   |
1  | #![deny(unused_must_use)]
   |         ^^^^^^^^^^^^^^^
